lib/vtls/openssl.c

```diff
@@ -2356,9 +2356,25 @@ static CURLcode get_cert_chain(struct connectdata *conn,
     if(!pubkey)
       infof(data, "   Unable to load public key\n");
     else {
-      switch(pubkey->type) {
+      int pktype;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
+    !defined(LIBRESSL_VERSION_NUMBER)
+      pktype = EVP_PKEY_id(pubkey);
+#else
+      pktype = pubkey->type;
+#endif
+      switch(pktype) {
       case EVP_PKEY_RSA:
-        BIO_printf(mem, "%d", BN_num_bits(pubkey->pkey.rsa->n));
+      {
+        RSA *rsa;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
+    !defined(LIBRESSL_VERSION_NUMBER)
+        rsa = EVP_PKEY_get0_RSA(pubkey);
+        BN_print(mem, rsa->n);
```

Comment:
@bagder But why are we calling BN_print directly to print n now instead of BN_num_bits like the old way below? Does that do the same thing because it doesn't look like it. Isn't `n` printed a few lines down `print_pubkey_BN(rsa, n, i);`? I don't have a working OpenSSL 1.1.0 setup but that looks inconsistent, at that point shouldn't we be retrieving the number of bits so when it's output it looks like for example:
RSA Public Key:2048
2048 being the number of bits in this case